fix error check when adding IZeroClampSeries

oruebel opened the pull request:
## Motivation

Fix #1409 


## Checklist

- [X] Did you update CHANGELOG.md with your changes?
- [X] Have you checked our [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [X] Have you ensured the PR clearly describes the problem and the solution?
- [X] Is your contribution compliant with our coding style? This can be checked running `flake8` from the source directory.
- [X] Have you checked to ensure that there aren't other open [Pull Requests](https://github.com/NeurodataWithoutBorders/pynwb/pulls) for the same change?
- [X] Have you included the relevant issue number using "Fix #XXX" notation where XXX is the issue number? By including "Fix #XXX" you allow GitHub to close issue #XXX when the PR is merged.

oruebel committed:
Fix #1409 fix error check when adding IZeroClampSeries

oruebel committed:
Update changelog

oruebel commented:
@rly this PR should be part of the next bug-fix release for PyNWB. Please update the target for the PR in case this needs  be merged with a different branch than ``dev``.

rly committed:
Merge branch 'dev' into fix/izero_clamp_add

rly commented:
Looks good to me. `dev` is the correct branch for bug fixes / changes that will be part of the next bug-fix release.

oruebel commented on tests/unit/test_icephys_metadata_tables.py:
```diff
@@ -255,6 +255,26 @@ def test_warn_if_IZeroClampSeries_with_stimulus(self):
                 id=np.int64(10)
             )
 
+    def test_noerror_if_IZeroClampSeries_with_stimulus(self):
```
```suggestion
    def test_noerror_if_IZeroClampSeries(self):
```

oruebel committed:
Update tests/unit/test_icephys_metadata_tables.py

oruebel commented:
@rly can you approve again. I noticed that the test I added was named incorrectly. I changed the name of the test to reflect what is actually being tested.
